Doc update

johnkim76 opened the pull request.

Commit:
update README.md

Commit:
add Index to README.md

fabianvf commented on README.md:
````diff
@@ -1,2 +1,163 @@
+USER apb
+```
+
+#### Spec File (`apb.yml`)
````
Might want to include the full path to the files for these

johnkim76 replied:
thanks.. will update

cfchase commented on README.md:
````diff
@@ -1,2 +1,163 @@
+            └── main.yml
+```
+### Directory Structure with Optional Files
+The directory structure with `optional/additional` files, may look like this:
````
nitpick: was there a reason this `optional/additional` looks like a code block?

johnkim76 replied:
thanks, will update

cfchase commented:
Minor nitpick issues, aside, it looks good to me.

ACK

Commit:
update to address comments in PR
- added links from the 'hello-world' apb to the example files
- formatting update
